runtime/compiler/control/OptionsPostRestore.cpp

```diff
@@ -150,7 +154,19 @@ J9::OptionsPostRestore::iterateOverExternalOptions()
 
          case J9::ExternalOptions::XcompilationThreads:
             {
-            // Need to allocate more if needed
+            char *compThreadsOption = J9::Options::_externalOptionStrings[J9::ExternalOptions::XcompilationThreads];
+            int32_t argIndex = FIND_AND_CONSUME_RESTORE_ARG(EXACT_MEMORY_MATCH, compThreadsOption, 0);
+            if (argIndex >= 0)
+               {
+               UDATA numCompThreads;
+               IDATA ret = GET_INTEGER_VALUE_RESTORE_ARGS(argIndex, compThreadsOption, numCompThreads);
+
+               if (ret == OPTION_OK && numCompThreads > 0)
+                  {
+                  TR::Options::_numUsableCompilationThreads = numCompThreads;
```

Comment:
Should we check that it does not exceed the number of allocated threads before checkpoint?